src/mscorlib/shared/System/Memory.cs

```diff
@@ -338,6 +328,15 @@ public bool TryGetArray(out ArraySegment<T> arraySegment)
                 arraySegment = new ArraySegment<T>(arr, _index, _length);
                 return true;
             }
+            else if (_length == 0)
```

Comment:
With "else", this won't kick in for an owned memory that doesn't wrap an array. Is that what's desired?